Source/AwsCommonRuntimeKit/mqtt/Mqtt5Options.swift

```diff
@@ -547,42 +547,34 @@ public class NegotiatedSettings {
         self.clientId = clientId
     }
 
-    static func convertFromNative(_ from: UnsafePointer<aws_mqtt5_negotiated_settings>?) -> NegotiatedSettings? {
-
-        guard let from else {
-            return nil
-        }
-
-        let _negotiatedSettings = from.pointee
-        let negotiatedMaximumQos = QoS(_negotiatedSettings.maximum_qos)
-        let negotiatedSessionExpiryInterval: TimeInterval = TimeInterval(_negotiatedSettings.session_expiry_interval)
-        let negotiatedReceiveMaximumFromServer = _negotiatedSettings.receive_maximum_from_server
-        let negotiatedMaximumPacketSizeToServer = _negotiatedSettings.maximum_packet_size_to_server
-        let negotiatedTopicAliasMaximumToServer = _negotiatedSettings.topic_alias_maximum_to_server
-        let negotiatedTopicAliasMaximumToClient = _negotiatedSettings.topic_alias_maximum_to_client
-        let negotiatedServerKeepAlive: TimeInterval = TimeInterval(_negotiatedSettings.server_keep_alive)
-        let negotiatedRetainAvailable = _negotiatedSettings.retain_available
-        let negotiatedWildcardSubscriptionsAvailable = _negotiatedSettings.wildcard_subscriptions_available
-        let negotiatedSubscriptionIdentifiersAvailable = _negotiatedSettings.subscription_identifiers_available
-        let negotiatedSharedSubscriptionsAvailable = _negotiatedSettings.shared_subscriptions_available
-        let negotiatedRejoinedSession = _negotiatedSettings.rejoined_session
-        let negotiatedClientId = _negotiatedSettings.client_id_storage.toString()
-
-        let negotiatedSettings = NegotiatedSettings(
-            maximumQos: negotiatedMaximumQos,
-            sessionExpiryInterval: negotiatedSessionExpiryInterval,
-            receiveMaximumFromServer: negotiatedReceiveMaximumFromServer,
-            maximumPacketSizeToServer: negotiatedMaximumPacketSizeToServer,
-            topicAliasMaximumToServer: negotiatedTopicAliasMaximumToServer,
-            topicAliasMaximumToClient: negotiatedTopicAliasMaximumToClient,
-            serverKeepAlive: negotiatedServerKeepAlive,
-            retainAvailable: negotiatedRetainAvailable,
-            wildcardSubscriptionsAvailable: negotiatedWildcardSubscriptionsAvailable,
-            subscriptionIdentifiersAvailable: negotiatedSubscriptionIdentifiersAvailable,
-            sharedSubscriptionsAvailable: negotiatedSharedSubscriptionsAvailable,
-            rejoinedSession: negotiatedRejoinedSession,
-            clientId: negotiatedClientId)
-
-        return negotiatedSettings
+    internal convenience init(_ from: UnsafePointer<aws_mqtt5_negotiated_settings>){
```

Comment:
If you make it an actual init instead of a convenience init, you won't need 20 different variables and can just do self.negotiatedSettings = etc, plus it is internal by default.

Reply:
I felt like it might be clearer intent to explicitly call it internal. I'm also calling the existing init in case we ever change anything there so we know to change it in both as a precaution. I'd rather do the assigning in a singular place but am open to doing a manual assign and not make it call itself if that makes more sense.

Reply:
Maybe if it's making a copy and then using the copy to call init which will then make another copy, that's bad and we should just assign it instead of it being convenience. Was planning on looking into how memory is handled across all these as next task.

Reply:
The init function is just 1 on 1 assignment, so I would not expect it to change in the future. If we ever add new properties to this class, Swift provides a compile time grantee that we will initialize it in both places. I think keeping it separate makes it more cleaner.

Reply:
Also it will make sure that we don't forget setting any new properties in second init function. With current approach, if we add a new property the convenience init will just start assigning that to NULL but if they are separate we will get a compile time error to handle it in the second init function as well.

Reply:
Just saw that the public init requires each member to be passed in so we won't run into the problem of forgetting to set it in the init function. However, we should probably assign some default to these members; otherwise, using it will be very cumbersome to have to pass each parameter individually.

Reply:
Looking into it further, I think I'm going to separate the inits and not make the native->swift one a convenience init.

Negotiated Settings is only ever created by the CRT and all fields must be present in the negotiated settings resulting from a connack to the server. There should be no default values and everything is to be explicitly set by the time it reaches here.

Reply:
> Negotiated Settings is only ever created by the CRT 
Then, we should make the other init internal as well.